sdk/ml/azure-ai-ml/azure/ai/ml/entities/_util.py

```diff
@@ -411,3 +414,37 @@ def normalize_job_input_output_type(input_output_value):
             input_output_value["job_input_type"] = FEB_JUN_JOB_INPUT_OUTPUT_TYPE_MAPPING[job_input_type]
         if job_type and job_type in FEB_JUN_JOB_INPUT_OUTPUT_TYPE_MAPPING:
             input_output_value["type"] = FEB_JUN_JOB_INPUT_OUTPUT_TYPE_MAPPING[job_type]
+
+
+def get_type_from_spec(data: dict, *, valid_keys: Iterable[str]) -> str:
+    """Get the type of the node or component from the yaml spec.
+    Yaml spec must have a key named "type" and exception will be raised if it's not once of valid_keys.
+
+    If internal components are enabled, related factory and schema will be updated.
+    """
+    _type, _ = extract_label(data.get(CommonYamlFields.TYPE, None))
+    schema = data.get(CommonYamlFields.SCHEMA, None)
+
+    from azure.ai.ml._utils.utils import try_enable_internal_components
```

Comment:
why is this function not under _internal namepsace?

Reply:
internal components will automatically be enabled once we implicitly import _internal namespace, so this `try_enable` must be put outside _internal namespace.